lib/syskit/deployment.rb

```diff
@@ -210,7 +211,11 @@ def create_deployed_task(
                       "expected #{base_syskit_task_model} or one of its subclasses"
             end
 
-            plan.add(task = syskit_task_model.new(orocos_name: mapped_name))
+            is_read_only_model = read_only.to_a.any? do |m|
+                mapped_name.include?(m)
+            end
```

Comment:
`mapped_name` is just a name. This is `read_only.include?(mapped_name)`

Reply:
The idea behind using it like this is to allow setting a deployment as read_only using only a substring. 

Let's say there is a gazebo model named `vessel` on the world, you can set all of the model's links, plugins, etc as read_only by just setting read_only: ["vessel"]. To set everything of the world as read_only, use ["gazebo"], for example. This way you dont need to express all the links, joints and plugins as read_only.

Using `read_only.include?(mapped_name)` would return true only if the exact string is present on the array.

Reply:
I am thinking this can be improved by adding another argument like `read_only_exact_match: true`. This way the user would have control whether to match the exact word or the substring.